operator/v1/types_dns.go

```diff
@@ -0,0 +1,101 @@
+package v1
+
+import (
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// +genclient
+// +genclient:nonNamespaced
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+// DNS manages the CoreDNS component to provide a name resolution service
+// for pods and services in the cluster.
+//
+// This supports the DNS-based service discovery specification:
+// https://github.com/kubernetes/dns/blob/master/docs/specification.md
+//
+// More details: https://kubernetes.io/docs/tasks/administer-cluster/coredns
+type DNS struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+	// spec is the specification of the desired behavior of the DNS.
+	Spec DNSSpec `json:"spec,omitempty"`
+	// status is the most recently observed status of the DNS.
+	Status DNSStatus `json:"status,omitempty"`
+}
+
+// DNSSpec is the specification of the desired behavior of the DNS.
+type DNSSpec struct {
+}
+
+// DNSStatus defines the observed status of the DNS.
+type DNSStatus struct {
+	// ClusterIP is the service IP through which this DNS is made available.
+	//
+	// In the case of the default DNS, this will be a well known IP that is used
+	// as the default nameserver for pods that are using the default ClusterFirst DNS policy.
+	//
+	// In general, this IP can be specified in a pod's spec.dnsConfig.nameservers list
+	// or used explicitly when performing name resolution from within the cluster.
+	// Example: dig foo.com @<service IP>
+	//
+	// More info: https://kubernetes.io/docs/concepts/services-networking/service/#virtual-ips-and-service-proxies
+	ClusterIP string `json:"clusterIP"`
+
+	// ClusterDomain is the local cluster DNS domain suffix for DNS services.
+	// This will be a domain name as defined in RFC 1034 (A-Z, a-z, 0-9 and the hyphen)
+	// Example: "cluster.local"
+	//
+	// More info: https://kubernetes.io/docs/concepts/services-networking/dns-pod-service
+	ClusterDomain string `json:"clusterDomain"`
+
+	// Conditions represent the latest available observations of a
+	// DNS's current state.
+	//
+	// +patchMergeKey=type
+	// +patchStrategy=merge
+	// +optional
+	Conditions []DNSCondition `json:"conditions,omitempty" patchStrategy:"merge" patchMergeKey:"type"`
```

Comment:
During the Ingress API experience I learned you must document the valid values and their meanings on this field, as the constants below aren't cross-referenced or otherwise represented in the generated docs.